FirebaseAuth/Sources/Auth/FIRAuth.m

```diff
@@ -2169,8 +2169,8 @@ - (nullable NSString *)getUserID {
 
 #pragma mark - Keychain sharing
 
-- (BOOL)useUserAccessGroup:(NSString *_Nullable)accessGroup
-                     error:(NSError *_Nullable *_Nullable)outError {
+- (BOOL)internalUseUserAccessGroup:(NSString *_Nullable)accessGroup
```

Comment:
Line 532 needs to change to use the internal version and you're done.

https://github.com/firebase/firebase-ios-sdk/blob/dba4325fd26b4a40a94339b425e6297e4a4f1f96/FirebaseAuth/Sources/Auth/FIRAuth.m#L532

The test needs to set up a stored user access group to effectively verify that the deadlock during initialization isn't happening.